service/api_task.go

```diff
@@ -470,41 +473,56 @@ func (as *APIServer) transitionIntentHostToDecommissioned(ctx context.Context, h
 	return nil
 }
 
+type agentExitParams struct {
+	host          *host.Host
+	remoteAddr    string
+	ec2InstanceID string
+}
+
 // prepareHostForAgentExit prepares a host to stop running tasks on the host.
 // For a quarantined host, it shuts down the agent and agent monitor to prevent
 // it from running further tasks. This is especially important for quarantining
 // hosts, as the host may continue running, but it must stop all agent-related
 // activity for now. For a terminated host, the host should already have been
 // terminated but is nonetheless alive, so terminate it again.
-func (as *APIServer) prepareHostForAgentExit(ctx context.Context, h *host.Host, remoteAddr string) (shouldExit bool, err error) {
-	switch h.Status {
+func (as *APIServer) prepareHostForAgentExit(ctx context.Context, params agentExitParams) (shouldExit bool, err error) {
+	switch params.host.Status {
 	case evergreen.HostQuarantined:
 		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
 		defer cancel()
-		if err := h.StopAgentMonitor(ctx, as.env); err != nil {
+		if err := params.host.StopAgentMonitor(ctx, as.env); err != nil {
 			return true, errors.Wrap(err, "stopping agent monitor")
 		}
 
-		if err := h.SetNeedsAgentDeploy(true); err != nil {
+		if err := params.host.SetNeedsAgentDeploy(true); err != nil {
 			return true, errors.Wrap(err, "marking host as needing agent or agent monitor deploy")
 		}
 
 		return true, nil
 	case evergreen.HostTerminated:
-		// The host should already be terminated but somehow it's still alive in
-		// the cloud. If the host was terminated very recently, it may simply
-		// need time for the cloud instance to actually be terminated and fully
-		// cleaned up. Beyond this short grace period, there's likely a state
-		// mismatch where Evergreen has marked the host terminated but the host
-		// has clearly not been terminated in the cloud.
-		if time.Now().Sub(h.TerminationTime) > 10*time.Minute {
-			grip.Error(message.Fields{
+		// The host should already be terminated but somehow the agent is still
+		// alive in the cloud. If the host was terminated very recently, it may
+		// simply need time for the cloud instance to actually be terminated and
+		// fully cleaned up. If this message logs, that means there is a
+		// mismatch between either:
+		// 1. What Evergreen believes is this host's state (terminated) and the
```

Comment:
It seems like there is a third case, which many not be a bug, and is described above as `If the host was terminated very recently, it may simply need time for the cloud instance to actually be terminated and fully cleaned up.` Should that be added to this list?

Reply:
The two cases I mentioned are cases when it would be logged to Splunk. The "terminated recently" case would not cause it to log because we give the host the 10-minute grace period to actually go away.

Reply:
Ah I missed that, thanks!